src/core/tsi/transport_security.c

```diff
@@ -228,6 +228,18 @@ tsi_result tsi_handshaker_result_extract_peer(const tsi_handshaker_result *self,
   return self->vtable->extract_peer(self, peer);
 }
 
+tsi_result tsi_handshaker_result_create_zero_copy_protector(
+    const tsi_handshaker_result *self, size_t *max_protected_frame_size,
+    tsi_zero_copy_protector **protector) {
+  if (self == NULL || protector == NULL) return TSI_INVALID_ARGUMENT;
+  if (self->vtable == NULL ||
```

Comment:
`self->vtable == NULL` is probably a `TSI_INVALID_ARGUMENT`. However, looks like this is implemented this way everywhere so we could open an issue to fix them all.